cmd/infracost/run.go

```diff
@@ -373,23 +420,75 @@ func runProjectConfig(cmd *cobra.Command, runCtx *config.RunContext, ctx *config
 			}
 
 			if e, ok := err.(*apiclient.APIError); ok {
-				return projects, fmt.Errorf("%v\n%s", e.Error(), "We have been notified of this issue.")
+				return nil, fmt.Errorf("%v\n%s", e.Error(), "We have been notified of this issue.")
 			}
 
-			return projects, err
+			return nil, err
 		}
 
 		schema.CalculateCosts(project)
 		project.CalculateDiff()
 	}
 
+	t2 := time.Now()
+	taken := t2.Sub(t1).Milliseconds()
+	ctx.SetContextValue("tfRunTimeMs", taken)
+
+	// wait for the hcl provider to finish if it hasn't already
+	wg.Wait()
+
 	spinner.Success()
+	out.projects = projects
 
 	if !runCtx.Config.IsLogging() {
 		cmd.PrintErrln()
 	}
 
-	return projects, nil
+	return out, nil
+}
+
+func runHCLProvider(wg *sync.WaitGroup, ctx *config.ProjectContext, usageFile *usage.UsageFile, runCtx *config.RunContext, out *projectOutput) {
+	defer func() {
+		err := recover()
+		if err != nil {
+			log.Debugf("recovered from hcl provider panic %s", err)
```

Comment:
by upstream you mean mixpanel right? There's no easy way to do this as currently this is outside of the current "run". We could have a successful `infracost-run` and this would fail. So I'm unsure what's the best way to do this. We'd maybe have to introduce a new event that we can report?

Reply:
discussed and we'll add a `infracost-error` event

Reply:
done in: https://github.com/infracost/infracost/pull/1378/commits/9a5b5de5b69683e4ff8c99afaa43d02d2d272975